### PR TITLE
Add a maintainers file

### PR DESCRIPTION
**Pull request by HeisSpiter**
This is a proposal so that ReactOS also gets a MAINTAINERS file.
The purpose of this file is to allow newcomers to know at a glance who's responsible of what in the project. It also puts some limits on what a maintainer is (and isn't).
It also helps newcomers knowing that some code is 3rd party and shouldn't directly be handled in ReactOS.
This is a pull request for comments, and approval.
Feel free to directly push to the branch what areas you want to be maintainer of.

Some examples:
- https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/MAINTAINERS
- https://source.winehq.org/git/wine.git/blob/HEAD:/MAINTAINERS

### CODE REVIEW

File: MAINTAINERS [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+There should be one and only one primary maintainer per area.
+
+In case of 3rd party code (also refered as upstream), the maintainer
+is responsible of updating periodically the source code and of
+managing local patches. He is not here to upstream code on your behalf.
+As responsible, he may refuse a local patch if you did not try to
+upstream your changes.
+
+This file uses a similar format to the Linux kernel MAINTAINERS file.
+Descriptions of section entries used here:
```
[the remark this answers is not in the document]
Comment (learn-more) [reply MAINTAINERS]: Link to our 'people of reactos' wiki page?

[the remark this answers is not in the document]
Comment (HeisSpiter) [reply MAINTAINERS]: I'd rather have everything in one file. Furthermore, our people of reactos page is not really maintained itself.

[the remark this answers is not in the document]
Comment (learn-more) [reply MAINTAINERS]: Fair enough

**Comment by Mouvedia** [comment]
Who is "hater" in this list?

**Comment by HeisSpiter** [comment]
I wonder whether we should enforce that a maintainer must have push access whereas this is not mandatory for a reviewer.
Any opinion?

**Comment by ThFabba** [comment]
Yes, I would argue that's the way to go. Perhaps also formalize what we've been doing: that people only add themselves rather than getting added by others.

**Comment by learn-more** [comment]
Push access seems mandatory.
As for only adding yourself, if there is some public record of someone asking to add himself that seems fine as well?

**Comment by HeisSpiter** [comment]
Yes, such tool exists: https://github.com/torvalds/linux/blob/master/scripts/get_maintainer.pl
This is specific to Linux, though. @ThFabba announced on IRC he was writing one for ReactOS.

**Pull request merged**